server.go

```diff
@@ -83,6 +86,10 @@ func init() {
 	internal.BinaryLogger = binaryLogger
 	internal.JoinServerOptions = newJoinServerOption
 	internal.RecvBufferPool = recvBufferPool
+
+	if testing.Testing() {
```

Comment:
seems like `testing.Testing()` was only made available since go.1.21.0 https://pkg.go.dev/testing#Testing

We maintain a backward compatibility to the last 3 versions of go. (i.e., v1.19+)

Reply:
Any suggestions what should I do about it? Options I can think of:
* add a hack to determine if this is a testing build, something like this https://stackoverflow.com/questions/14249217/how-do-i-know-im-running-within-go-test
* Don't enable the delay by default, add a server option to activate it
* Enable delay by default and ignore the fact that it could break some tests that expect the server to be shut-down faster then 1 sec. Add a server option so that users can disable (or maybe configure?) the delay in their tests.